### ISSUE TITLE
Autocorrect produces invalid syntax for TrailingCommaInArrayLiteral/TrailingCommaInHashLiteral with EnforcedStyleForMultiline: comma

### ISSUE THREAD

**Issue by amarshall** [rubocop-hq/rubocop#5932]
## Expected behavior

Autocorrect does not produce invalid Ruby.

```ruby
# frozen_string_literal: true

[
  { one: 'bar',
    bar: 'baz' },
  { one: 'bar',
    bar: 'baz' },
  [1,
   2],
  [3,
   4],
]
```

## Actual behavior

Autocorrect produces invalid Ruby.

```ruby
# frozen_string_literal: true

[
  { one: 'bar',
    bar: 'baz'},,
  { one: 'bar',
    bar: 'baz' },
  [1,
    2],,
  [3,
    4],
]
```

## Steps to reproduce the problem

Create the following files:

```yaml
# rubocop.yml
Style/TrailingCommaInArrayLiteral:
  EnforcedStyleForMultiline: comma
Style/TrailingCommaInHashLiteral:
  EnforcedStyleForMultiline: comma
```

```ruby
# comma-literal.rb
[
  { one: 'bar',
    bar: 'baz'
  },
  { one: 'bar',
    bar: 'baz'},
  [ 1,
    2
  ],
  [ 3,
    4
  ]
]
```

Run `rubocop -a`, then `ruby comma-literal.rb`, and witness that Ruby complains of invalid syntax:

```
comma-literal.rb:5: syntax error, unexpected ',', expecting ']'
    bar: 'baz'},,
                 ^
comma-literal.rb:7: syntax error, unexpected ',', expecting end-of-input
    bar: 'baz' },
```

## RuboCop version

```
$ rubocop -V
0.56.0 (using Parser 2.5.1.0, running on ruby 2.4.2 x86_64-darwin17)
```

**Comment by execat** [comment]
This might be also related to this: https://github.com/bbatsov/rubocop/issues/5669
They seems similar in the way that `Style/TrailingCommaIn<Something>Literal` creates an unexpected output by adding an invalid `,`.

**Comment by garettarrowood** [comment]
I'm stumped. I can duplicate this bug with this code by adding it to the appropriate place to the `cli_autocorrect_spec.rb`. I can then track `Layout/MultilineHashBraceLayout` removing the appropriate spaces, then get to the point in which `TrailingCommaInHashLiteral` adds it's comma.  However, even though those spaces have been deleted, the `processed_source` at this last step is not updated. 

So the `PunctuationCorrector` does not have the correct code available to make it intelligent enough to account for this scenario. I need to figure out exactly why the `processed_source` is not updated, then how to access this updated code to fix the issue.

**Comment by kbacha** [comment]
I have also run into this, and I agree that it seems like it's related to an interaction between two cops. This was my input code:

```ruby
# frozen_string_literal: true

def test(*)
end

hash = { a: 1,
  b: 2
}

hello = test("some string")
```

which then becomes:

```ruby
# frozen_string_literal: true

def test(*)
end

hash =
  {
    a: 1,
    b: 2,
  },

  hello = test("some string")
```

Here's the output:

```bash
$ rubocop -a test.rb
Inspecting 1 file
W

Offenses:

test.rb:6:1: W: Lint/UselessAssignment: Useless assignment to variable - hash.
hash =
^^^^
test.rb:6:7: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
hash =
      ^
test.rb:6:8: C: [Corrected] Layout/FirstArrayElementLineBreak: Add a line break before the first element of a multi-line array.
hash = {  ...
       ^^
test.rb:6:9: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
hash = {
        ^
test.rb:6:10: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
hash = { a: 1,
         ^^^^
test.rb:7:1: C: [Corrected] Layout/IndentAssignment: Indent the first line of the right-hand-side of a multi-line assignment.
{ ...
^
test.rb:7:1: C: [Corrected] Layout/IndentHash: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
a: 1,
^^^^
test.rb:7:3: C: [Corrected] Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
  b: 2
  ^^^^
test.rb:7:3: C: [Corrected] Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.
  b: 2
  ^^^^
test.rb:8:1: C: [Corrected] Layout/MultilineHashBraceLayout: Closing hash brace must be on the same line as the last hash element when opening brace is on the same line as the first hash element.
}
^
test.rb:8:10: C: [Corrected] Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
         b: 2},
         ^^^^
test.rb:8:10: C: [Corrected] Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.
         b: 2},
         ^^^^
test.rb:8:14: C: [Corrected] Layout/MultilineHashBraceLayout: Closing hash brace must be on the line after the last hash element when opening brace is on a separate line from the first hash element.
         b: 2},
             ^
test.rb:8:14: C: [Corrected] Layout/SpaceInsideHashLiteralBraces: Space inside } missing.
         b: 2},
             ^
test.rb:9:1: C: [Corrected] Layout/AlignHash: Align the elements of a hash literal if they span more than one line.
b: 2,
^^^^
test.rb:10:1: C: [Corrected] Layout/AlignArray: Align the elements of an array literal if they span more than one line.
hello = test("some string")
^^^^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:10:2: C: [Corrected] Layout/IndentHash: Indent the right brace the same as the start of the line where the left brace is.
 },
 ^
test.rb:11:1: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
test.rb:12:3: W: Lint/UselessAssignment: Useless assignment to variable - hello.
  hello = test("some string")
  ^^^^^
test.rb:12:8: C: [Corrected] Layout/AlignArray: Align the elements of an array literal if they span more than one line.
       hello = test("some string")
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 20 offenses detected, 18 offenses corrected
```
